### DISCUSSION TITLE
Mode insensitive is not working

### DISCUSSION THREAD

**Discussion by ggukggukee** [prisma/prisma#13854] [Q&A]
### Bug description

Prisma with Postgres database not working mode: insensitive.

I have a string in User table with name column 'Alex' and if I search 'alex' with case insensitive mode - nothing is found.

### How to reproduce

1. Prisma findMany for name 'alex' with case insensitive mode
2. Return nothing

### Expected behavior

Return query with user named Alex

### Prisma information

schema.prisma
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id Int @id @default(autoincrement())
  login String?
  password String?
  email String?
  surname String
  name String
}
```

users search query
```js
router.get('/search', async (req, res) => {
  const query = req.query.query; // 'alex'
  try {
    const users = await prisma.user.findMany({
      where: {
        name: {
          startsWith: query,
          mode: 'insensitive'
        }
      },
    });

    return res.json(users);
  } catch (err) {
    console.log(err);
  }
});
```

### Environment & setup

- OS: Mac OS 12.1 
- Database: PostgreSQL 14
- Node.js version: 16.14.2 

### Prisma Version

```
prisma                  : 3.14.0
@prisma/client          : 3.14.0
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 2b0c12756921c891fec4f68d9444e18c7d5d4a6a (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 2b0c12756921c891fec4f68d9444e18c7d5d4a6a (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core 2b0c12756921c891fec4f68d9444e18c7d5d4a6a (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt 2b0c12756921c891fec4f68d9444e18c7d5d4a6a (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : 2b0c12756921c891fec4f68d9444e18c7d5d4a6a
Studio                  : 0.460.0
```

**Comment by austincrim** [comment]
Hey @ggukggukee 👋,

Thanks for all the info! I was unable to reproduce this on my own computer.

This query:
```ts
let result = await prisma.user.findMany({
  where: {
    name: {
      startsWith: 'alex',
      mode: 'insensitive'
    }
  }
})
```
returned this result:
```ts
[
  { id: 1, name: 'alex' },
  { id: 2, name: 'ALEX' },
  { id: 3, name: 'Alex' }
]
```

Have you double checked your existing data to make sure the expected data is in the table?

**Reply by ggukggukee** [reply]
Seems like it is not working with cyrillic letters.

**Reply by matthewmueller** [reply]
Did you check your collation settings? Also, any chance you know what the correct SQL would be?

**Reply by aqrln** [reply]
I can't reproduce it too, even with cyrillic letters.

```ts
  const users = await prisma.user.findMany({
    where: {
      name: {
        startsWith: 'кот',
        mode: 'insensitive',
      },
    },
  })
```

```
[
  { id: 1, name: 'кот' },
  { id: 2, name: 'Кот' },
  { id: 3, name: 'КОТ' }
]
```

**Reply by ggukggukee** [reply]
> Did you check your collation settings? Also, any chance you know what the correct SQL would be?

No. Can you tell me if you know about that

**Reply by pantharshit00** [reply]
Hello @ggukggukee 

Sorry for late follow up here. A collation basically defines rules for characters and their sorting order which are used in the database. You can read more about it here:https://www.postgresql.org/docs/current/collation.html

**Reply by pantharshit00** [reply]
Hello @ggukggukee,

Were you able to solve this? Please do let us know if this is still open.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
